### DISCUSSION TITLE
Achievements: don't make me click "Update preferences", just save when I click the checkbox

### DISCUSSION THREAD

**Discussion by rattrayalex** [community/community#18671] [Other Feature Feedback, Questions, & Ideas]
I don't like achievements; I'm offended that the first things someone should see about my profile are that I "YOLO" (sorry that I merged directly to a small private repo?) and have had _three_ PR's merged (I think I've had hundreds or thousands, but ok?).

I wanted to turn it off, and found [this](https://github.com/github-community/community/discussions/18587#discussioncomment-2948430) but if you toggle the checkbox and navigate away, it doesn't save, and doesn't warn you that you have unsaved changes. Meanwhile, other checkboxes in my github settings (eg, notifications) apply atomically and are saved right away, without needing to press a button to "commit" them.

For users who are upset that they've been decorated with flair, at least let them have a single click to remove it – imagine how more upset they'd be when they try to turn it off and it's still there. It feels intentionally deceptive, even though it's not.

Additionally, when viewing one's own profile, there should be a button to remove this – or at least navigate to the part of the settings page to remove it – right there. I shouldn't have to go googling.

EDIT: more color, because elsewhere someone said this "wasn't an issue because other UI's make you press a button to save changes":

Many other parts of _github's_ toggle-based settings save on commit, and there is no indication that your change hasn't been saved (no beforeunload handler, no changing of the button from disabled to active, etc). Even the wording "Update preferences" implied to me that it would open some other dialog about my preferences related to achievements (perhaps changing their styling or opting into certain badges or something).

**Comment by stevenxxiu** [comment]
Just a comment on how to block this from *uBlock Origin*:

    github.com##.d-block:has(a[href*="?tab=achievements"])
    github.com##.d-md-block:has(a[href*="?tab=achievements"])

The only answer is for *Microsoft* to remove this horrid idea.

**Comment by radar-blip** [comment]
You are kidding, right?
